### PR TITLE
Input modes

### LINKED ISSUE #382: Value input modes core
image

### LINKED ISSUE #383: Value input mode Windows
Windows implementation of #382 

Existing design:
image

### PR DESCRIPTION
**Pull request by ermau**
Closes #382, closes #383

### CODE REVIEW

File: Xamarin.PropertyEditing.Tests/MockControls/MockSampleControl.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ public MockSampleControl ()
 			AddProperty<int> ("UnsetInteger", ReadWrite, valueSources: ValueSources.Local);
 			AddProperty<float> ("FloatingPoint", ReadWrite);
 			AddProperty<string> ("String", ReadWrite, valueSources: ValueSources.Local | ValueSources.Resource | ValueSources.Binding);
+			AddProperty<int> ("Width", ReadWrite, valueSources: ValueSources.Local | ValueSources.Resource | ValueSources.Binding, inputModes: new[] { new InputMode("Auto", true), new InputMode("Star"), new InputMode("Pixel"), });
```
Comment (bleroy) [inline Xamarin.PropertyEditing.Tests/MockControls/MockSampleControl.cs]: Either those are going to surface to the user, and they need to be localizable, or they're magic strings, and could probably be replaced with an enum.

File: Xamarin.PropertyEditing/ViewModels/NumericPropertyViewModel.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
```
Comment (bleroy) [inline Xamarin.PropertyEditing/ViewModels/NumericPropertyViewModel.cs]: Oops?

File: Xamarin.PropertyEditing.Tests/MockControls/MockSampleControl.cs (hunk 1 above)
Comment (ermau) [reply Xamarin.PropertyEditing.Tests/MockControls/MockSampleControl.cs]: They're up to the provider to provide. In theory they will likely correspond to fixed names from code (enums, const properties) that should not be translated.

Comment (bleroy) [reply Xamarin.PropertyEditing.Tests/MockControls/MockSampleControl.cs]: Ah yes, this is just the mock implementation, indeed.

**Pull request merged**